service/src/main/scala/za/co/absa/loginsvc/rest/config/JwtConfig.scala

```diff
@@ -22,9 +22,29 @@ import za.co.absa.loginsvc.rest.config.validation.ConfigValidationResult.{Config
 import scala.util.{Failure, Success, Try}
 
 case class JwtConfig(
-  algName: String,
-  expTime: Int
-) extends ConfigValidatable {
+                      generateInMemory: Option[GenerateKeysConfig],
+                      fetchFromAws: Option[FetchSecretConfig])
+  extends ConfigValidatable {
```

Comment:
Why not just make `JwtConfig` a `trait` with both subtypes implementing it? Is there a reason why this is not feasible that I am missing?

Reply:
Implemented something along these lines